src/allotropy/parsers/agilent_gen5/agilent_gen5_structure.py

```diff
@@ -540,10 +570,10 @@ def _get_sources(
     return sources or measurements
 
 
-def _create_metadata(header_data: HeaderData, read_data: ReadData) -> Metadata:
+def _create_metadata(header_data: HeaderData, read_data: list[ReadData]) -> Metadata:
     return Metadata(
         device_type=DEVICE_TYPE,
-        detection_type=read_data.read_mode.value,
+        detection_type=read_data[0].read_mode.value,
```

Comment:
This doesn't make sense to set here, since the detection type does not apply to all measurements.

You handle this in the schema mapper by hard-coding the read mode, which is fine, but it means this is not used.

Please remove this, and make `detection_type` optional in the `Metadata` class

Reply:
Agree, this field should be removed since we support multi read mode and we take it from the measurement.